Update MemoryFence definition logic

AjanZhong opened the pull request:
Both nvme and virtual_media driver invokes MemoryFence function,
implement inline function in corresponding header files as inline
function to avoid multi definition link issue.

Also provide MemoryFence symbol in libefiwrapper for kernelflinger
to link MemoryFence symbol which supposed to be implemented in EFI
componment.

Tracked-On: OAM-113553

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed Engineering Build for this issue, build is FAILURE. Please check the linked Tracked-On issue/Android CI Web  for more details

yang8621 committed:
Update MemoryFence definition logic

Both nvme and virtual_media driver invokes MemoryFence function,
implement inline function in corresponding header files as inline
function to avoid multi definition link issue.

Also provide MemoryFence symbol in libefiwrapper for kernelflinger
to link MemoryFence symbol which supposed to be implemented in EFI
componment.

Tracked-On: OAM-113553
Signed-off-by: Zhong,Fangjian <fangjian.zhong@intel.com>

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:Update Bugs And Epics Update Bugs And Epics

More details, logs, and binaries: /absp/builders/celadon_smr0_master-preintegration/builds/1060

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:Update Bugs And Epics Update Bugs And Epics

More details, logs, and binaries: /absp/builders/celadon_smr0_master-preintegration/builds/1062

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckBug Bad comments/Bugs

More details, logs, and binaries: /absp/builders/celadon_smr0_master-preintegration/builds/1075

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:Trigger Builds Triggered: build-cels - r1 (celadon_ivi), build-cels - r2 (r2_apollo_ivi), build-cels - r2 (r2_blizzard_ivi) - extra, build-cels - r2 (r2_celadon_ivi), build-cels - r2 (r2_everest_ivi)

More details, logs, and binaries: /absp/builders/celadon_smr0_master-preintegration/builds/1076

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed Engineering Build for this issue, build is FAILURE. Please check the linked Tracked-On issue/Android CI Web  for more details

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed Engineering Build for this issue, build is FAILURE. Please check the linked Tracked-On issue/Android CI Web  for more details

sysopenci commented:
Android CI has completed Engineering Build for this issue, build is FAILURE. Please check the linked Tracked-On issue/Android CI Web  for more details

sysopenci commented:
Android CI has completed Engineering Build for this issue, build is FAILURE. Please check the linked Tracked-On issue/Android CI Web  for more details

sysopenci commented:
Android CI has completed Engineering Build for this issue, build is SUCCESS.To merge the changes please click on APPROVE FOR MERGE button in Android CI WEB

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckBug Bad comments/Bugs

More details, logs, and binaries: /absp/builders/celadon_smr0_master-preintegration/builds/1088

sysopenci reviewed (commented):
PR integrated into 'preintegration' builder



More details, logs, and binaries: /absp/builders/celadon_smr0_master-preintegration/builds/1089

Done by "Zhong, Fangjian" <fangjian.zhong@intel.com>

sysopenci reviewed (commented):
PR integrated into 'premerge' builder



More details, logs, and binaries: /absp/builders/celadon_smr0_master-premerge/builds/587

Done by "Zhong, Fangjian" <fangjian.zhong@intel.com>

sysopenci reviewed (commented):
cactus finished building and testing your changes 

It was requested for merge by "Zhong, Fangjian" <fangjian.zhong@intel.com> 

so we merge it. Thanks for your contribution.

more details, logs, and binaries: /absp/builders/celadon_smr0_master-mergerequest/builds/606
